server/command_handlers.py

```diff
@@ -0,0 +1,44 @@
+import random
+from collections.abc import Awaitable
+from collections.abc import Callable
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from server.repositories.sessions import Session
+
+CommandHandler = Callable[["Session", list[str]], Awaitable[str | None]]
+
+command_handlers: dict[str, CommandHandler] = {}
```

Comment:
our if/else cascade becomes this, where keys are the conditions (`trigger`) and values are the command handling functions defined below (e.g. `echo_handler`, `roll_handler`, etc.).

the functions are added to this mapping the via decorator `@command_handler` - specifically on line 20.